draft-ietf-jsonpath-base.md

```diff
@@ -1521,19 +1521,21 @@ A function expression is well-typed if all of the following are true:
   to have a result type of `LogicalType`, `SingleNodeType`, or `NodesType`.
 * If it occurs directly as a `comparable` in a comparison, the
   function is declared to have a result type of `ValueType` or `SingleNodeType`.
-* Otherwise, it occurs as an argument to a further function
-  expression and the following rules for function arguments apply to
-  its declared result type.
 * Each argument of the function conforms to the declared type of the corresponding declared
-  parameter according to one of the following rules:
-   * The argument is a function expression with declared result type that is the same as
-     the declared type of the parameter.
-   * The argument is a function expression with declared result type `SingleNodeType` and
-     the declared type of the parameter is `NodesType`.
-   * The argument is a literal primitive value and the declared type of the parameter is `ValueType`.
-   * The argument is a Singular Path and the declared type of the parameter is `SingleNodeType`.
-   * The argument is a Singular Path or `filter-path` (which includes
-     Singular Paths) and the declared type of the parameter is `NodesType`.
+  parameter according to one of the following scenarios:
+   * The declared type of the parameter is `ValueType` and the argument is one of:
+     * a function with declared type of `ValueType` or `SingleNodeType`
+     * a `singular-path`
+     * a `literal`
+   * The declared type of the parameter is `LogicalType` and the argument is one of:
+     * a function with declared type of `LogicalType`, `SingleNodeType`, or `NodesType`
+     * a `filter-path` (which includes `singular-path`)
```

Comment:
The PR needs to explain how to get from `SingleNodeType`, `NodesType`, and `filter-path` to `LogicalType`. C.f. the following text from "Existence Tests":
> A path by itself in a Logical context is an existence test which yields true if the path selects at least one node and yields false if the path does not select any nodes.

Reply:
The text on this branch actually reads:

> A path by itself in a Logical context is an existence test which yields true if the path selects at least one node and yields false if the path does not select any nodes.
> A function expression with result type `SingleNodeType` or `NodesType` in a Logical context is an existence test which yields true if the function returns a non-empty nodelist and yields false if the function returns an empty node-list.

so it's already there.

Reply:
> supplying an argument to a parameter with type `LogicalType` has any connection with a "logical context"

I don't understand.  How can anywhere `LogicalType` is used _not_ be a "logical context?"

Reply:
Because we haven't actually defined "logical context", so it's somewhat open to interpretation. My interpretation in the context of the existence test section is that "logical context" refers to "logical-expr":
```
logical-expr        = logical-or-expr
logical-or-expr     = logical-and-expr *(S "||" S logical-and-expr)
                        ; disjunction
                        ; binds less tightly than conjunction
logical-and-expr    = basic-expr *(S "&&" S basic-expr)
                        ; conjunction
                        ; binds more tightly than disjunction

basic-expr          = paren-expr /
                      comparison-expr /
                      test-expr

```

Reply:
Addressed in [ba8c16b](https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/418/commits/ba8c16bf51f0aa495c605ca16e46006a1dbb3340)

Reply:
Yes, that does the job nicely.